pdns/pdns_recursor.cc

```diff
@@ -2409,6 +2411,12 @@ static void makeThreadPipes()
     tps.readFromThread = fd[0];
     tps.writeFromThread = fd[1];
 
+    if(pipe(fd) < 0)
+      unixDie("Creating pipe for inter-thread communications");
+    tps.readQueriesToThread = fd[0];
+    tps.writeQueriesToThread = fd[1];
+    setNonBlocking(tps.writeQueriesToThread);
```

Comment:
Would it be a problem if `setNonBlocking()` would fail ?

Reply:
It would not really be an issue since we would just be in the same situation than we were previously but I'll add a check nevertheless!